modules/signals/src/signal-store.ts

```diff
@@ -1355,12 +1355,18 @@ export function signalStore(
         getInitialInnerStore()
       );
       const { stateSignals, props, methods, hooks } = innerStore;
-      const storeMembers = { ...stateSignals, ...props, ...methods };
+      const storeMembers: Record<string | symbol, unknown> = {
+        ...stateSignals,
+        ...props,
+        ...methods,
+      };
 
       (this as any)[STATE_SOURCE] = innerStore[STATE_SOURCE];
 
-      for (const key in storeMembers) {
-        (this as any)[key] = storeMembers[key];
+      for (const key of Reflect.ownKeys(storeMembers)) {
+        if (typeof key === 'string' || typeof key === 'symbol') {
```

Comment:
Can we also include `number`s?
Because these can also be used as property names.

Or maybe this isn't a problem?

Reply:
Short answer: No, and I've added a test as well.

In my understanding, numbers are always serialized during the runtime. That's why Reflect.ownKeys returns 'symbol | string`.

Example

```typescript
const o = {1: 'a'}
o[1]; // returns 'a'
o['1']; // returns 'a'
```

The if condition was necessary because of the TypeScript compiler. It doesn't accept a union type.